### PR TITLE
SI-9813: Add Java 1.6. and 1.8 java.lang.Math methods into scala.math.package

### PR DESCRIPTION
**Pull request by janekdb**
## TODO
- [x] Apply review comments from #5314 
- [x] Close #5314
- [x] Review group priorities
- [x] Remove `WIP` label
## Reviewing

The matching Scaladoc is here to make reviewing the groupings easier,

http://janekdb.github.io/scala/PR-5382/scala/math/index.html

All Java 6 & 8 math methods should be present. The assumption around the documentation is interested users will use the Oracle documentation.

Happy to take reviews from anyone with the inclination!

### CODE REVIEW

File: src/library/scala/math/package.scala [hunk 1]
```diff
@@ -215,9 +218,37 @@ package object math {
     */
   def signum(x: Long): Long     = java.lang.Long.signum(x)
   /** @group signum */
-  def signum(x: Float): Float   = java.lang.Math.signum(x)
+  def signum(f: Float): Float   = java.lang.Math.signum(f)
```
Comment (janekdb) [inline src/library/scala/math/package.scala]: Reverted.

File: src/library/scala/math/package.scala [hunk 2]
```diff
@@ -215,9 +218,37 @@ package object math {
...
   /** @group signum */
-  def signum(x: Double): Double = java.lang.Math.signum(x)
+  def signum(d: Double): Double = java.lang.Math.signum(d)
```
Comment (janekdb) [inline src/library/scala/math/package.scala]: Reverted.

File: src/library/scala/math/package.scala [hunk 3]
```diff
@@ -315,13 +356,83 @@ package object math {
   /** Returns the size of an ulp of the given `Double` value.
     * @group ulp
     */
-  def ulp(x: Double): Double = java.lang.Math.ulp(x)
+  def ulp(d: Double): Double = java.lang.Math.ulp(d)
```
Comment (janekdb) [inline src/library/scala/math/package.scala]: Reverted.

File: src/library/scala/math/package.scala [hunk 4]
```diff
@@ -315,13 +356,83 @@ package object math {
...
 
   /** Returns the size of an ulp of the given `Float` value.
     * @group ulp
     */
-  def ulp(x: Float): Float = java.lang.Math.ulp(x)
+  def ulp(f: Float): Float = java.lang.Math.ulp(f)
```
Comment (janekdb) [inline src/library/scala/math/package.scala]: Reverted.

**Comment by janekdb** [comment]
Review by @soc, @SethTisue , @som-snytt + you if you are reading this :)

**Comment by janekdb** [comment]
/rebuild

```
# There is insufficient memory for the Java Runtime Environment to continue.
```

**Comment by SethTisue** [comment]
https://github.com/scala/scala-jenkins-infra/issues/181 strikes again

**Comment by som-snytt** [comment]
We oughtn't accuse Jenkins of having an insufficient memory just because he's become a bit forgetful. Who of us has not these days?

Or by "strikes again", did you mean he actually went on strike? Is Jenkins really French? Jeanquins?

**Comment by janekdb** [comment]
@SethTisue - kindly remove `WIP` label.

**Comment by janekdb** [comment]
There are worse butlers than Jenkins so count your blessings

**Comment by janekdb** [comment]
Review by @ShaneDelmore if you like!

**Comment by janekdb** [comment]
Review by @som-snytt

**Comment by som-snytt** [comment]
LGTM

With the caveat that I don't know if this what number-crunchers really want and whether Scala.js enjoys uniform support and behavior. That's probably a separate issue about platforms, including things like regex flavor. There's also the other discussion about reduced API commitment because extension methods are trivial.

**Comment by sjrd** [comment]
[Scala.js supports most of those](https://github.com/scala-js/scala-js/blob/master/javalanglib/src/main/scala/java/lang/Math.scala), and the missing ones could be implemented.

**Comment by SethTisue** [comment]
well, it's a shame this didn't make 2.12.x :-(

we'll have to retarget this for 2.13 once a 2.13 branch exists

**Comment by adriaanm** [comment]
(Changes that break compatibility had to go in before the RC cycle started, as [announced on July 6](https://groups.google.com/forum/#!msg/scala-internals/hxDjZfm1h3c/Y5OkHdatAAAJ))

**Comment by som-snytt** [comment]
This would be an easy one to try over at TLS.

**Comment by SethTisue** [comment]
@janekdb there is now a 2.13.x branch, if you want to pick this back up

**Comment by janekdb** [comment]
Hmm, somehow I have managed to close this request instead of moving it to be for 2.13.x. Not my intention. Will try again later!

**Comment by som-snytt** [comment]
BTW, `TLS` === "The Latest Scala", of course... or maybe `Tigger Loves Scala`.

**Comment by janekdb** [comment]
TLS. Thanks for "decrypting" this. I was wondering!

**Comment by janekdb** [comment]
Moved to #5548

**Pull request closed without merging**